cpp/include/cuvs/neighbors/nn_descent.hpp

```diff
@@ -353,7 +362,17 @@ auto build(raft::resources const& res,
   -> cuvs::neighbors::nn_descent::index<uint32_t>;
 
 /**
+ * @brief Test if we have enough GPU memory to run NN descent algorithm.
+ *
+ * @param res
+ * @param dataset shape of the dataset
+ * @param idx_size the size of index type in bytes
+ * @return true if enough GPU memory can be allocated
+ * @return false otherwise
+ */
+bool has_enough_device_memory(raft::resources const& res,
```

Comment:
Small nitpick- can we move this to the `cuvs::neighbors::nn_descent::helpers` namespace please? 

We can do it in a follow-on PR for 24.08 so we can get this merged ASAP.